dask/dataframe/core.py

```diff
@@ -7918,7 +7926,13 @@ def repartition(df, divisions=None, force=False):
 def _reduction_chunk(x, aca_chunk=None, **kwargs):
     o = aca_chunk(x, **kwargs)
     # Return a dataframe so that the concatenated version is also a dataframe
-    return o.to_frame().T if is_series_like(o) else o
+    if is_series_like(o):
+        result = o.to_frame().T
+        if o.dtype.kind == "O":
```

Comment:
Not a fun to multiple returns. If they were quick returns, maybe.

Reply:
Sorry I don't understand?

Reply:
These DataFrame have only one line btw, so this should be super fast

Reply:
Sorry, this comment was not very clear. It's just a code style nitpick.

Usually it's considered a good pattern to have a single point where a function returns control to its enclosing scope. Your lines 7929-7935 would look something like this:

```python
    if is_series_like(o):
        result = o.to_frame().T
        if o.dtype.kind == "O":
            result = result.infer_objects()
    else:
        result = o
    return result
```

Sometimes, this pattern leads to too much indentation. So people use "quick returns" to break out early, which makes the code more flat and easier to read:

```python
    if not is_series_like(o):
        return o
    result = o.to_frame().T
    if not o.dtype.kind == "O":
        return result
    return result.infer_objects()
```

Your snippet is following neither style. It has multiple points of returning control, but isn't saving any indentation levels.

I hope this helps.

Reply:
personally I don't like the first style. Imo this makes it harder to follow (recognising whether the result is modified again, especially if the code is more complex).

I'd be ok with the second one

Reply:
Updated, also in the base PR